pkg/controller/zookeepercluster/zookeepercluster_controller.go

```diff
@@ -185,20 +188,71 @@ func (r *ReconcileZookeeperCluster) reconcileStatefulSet(instance *zookeeperv1be
 	} else if err != nil {
 		return err
 	} else {
-		r.log.Info("Updating StatefulSet",
-			"StatefulSet.Namespace", foundSts.Namespace,
-			"StatefulSet.Name", foundSts.Name)
-		zk.SyncStatefulSet(foundSts, sts)
-		err = r.client.Update(context.TODO(), foundSts)
-		if err != nil {
-			return err
+		foundSTSSize := *foundSts.Spec.Replicas
+		newSTSSize := *sts.Spec.Replicas
+		if newSTSSize < foundSTSSize {
+			// We're dealing with STS scale down
+			if !r.isConfigMapInSync(instance) {
+				r.log.Info("Skipping StatefulSet reconcile as ConfigMap not updated yet.")
+				return nil
```

Comment:
In the case that it is not reconciled, we exit, and I assume we retry this function eventually? what's the mechanism for retrying?

Reply:
Yes. We will retry it eventually.